### PR TITLE
chore(kinesis): convert tests to jest

### PR DESCRIPTION
**Pull request by shivlaks**
### Commit Message
chore(kinesis): convert tests to jest (#7314)

used statement and branch thresholds of `80` although this module is currently above 90 for both.

file needed to be renamed as our detection mechanism discovers `test*.js` as nodeunit files and `*test.ts` as jest files. 
### End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-kinesis/package.json [hunk 1]
```diff
@@ -60,14 +60,30 @@
     "url": "https://aws.amazon.com",
     "organization": true
   },
+  "jest": {
```
Comment (eladb) [inline packages/@aws-cdk/aws-kinesis/package.json:63]: Do we really need to duplicate all this config everywhere? Isn't there a way to use a default config? @RomainMuller ?

File: packages/@aws-cdk/aws-kinesis/package.json [hunk 2]
```diff
@@ -60,14 +60,30 @@
...
+  },
   "license": "Apache-2.0",
   "devDependencies": {
     "@aws-cdk/assert": "0.0.0",
-    "@types/nodeunit": "^0.0.30",
     "cdk-build-tools": "0.0.0",
     "cdk-integ-tools": "0.0.0",
     "cfn2ts": "0.0.0",
-    "nodeunit": "^0.11.3",
+    "jest": "^24.9.0",
```
Comment (eladb) [inline packages/@aws-cdk/aws-kinesis/package.json:86]: Is this required? Isn't `jest` a dependency of `cdk-test`?

File: packages/@aws-cdk/aws-kinesis/package.json (hunk 1 above)
Comment (iliapolo) [reply packages/@aws-cdk/aws-kinesis/package.json:63]: I believe we don't need this anymore, I know some modules simply have `"jest": {}` @nija-at?

Comment (shivlaks) [reply packages/@aws-cdk/aws-kinesis/package.json]: @iliapolo @eladb - you're right - we did centralize this - updated!

I think we still have a whole bunch of modules that have jest keys configured, as only about ~15 modules use this style. I'll clean em up in a separate PR

File: packages/@aws-cdk/aws-kinesis/package.json (hunk 2 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-kinesis/package.json]: good catch! it is not, cdk-build tools already bring all these needed dependencies in.

File: packages/@aws-cdk/aws-kinesis/package.json (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-kinesis/package.json]: https://github.com/aws/aws-cdk/pull/6869

**Pull request merged**